Support for Swift 5.6

daveverwer opened the pull request:
Hi Jhonatan!

I’d love to add both of these packages (SwiftSafeURL and 
XcodeIssueReporting) to the [Swift Package Index](https://swiftpackageindex.com). Unfortunately we currently don’t support Swift 5.7 packages as it’s still in beta. However, I checked by building them locally with 5.6 and I don’t think you’re using any 5.7 specific features?

Would you consider these PRs to reduce the tools version?

Thanks!

daveverwer committed:
Changed Swift tools version to 5.6.

baguio commented:
Hi Dave! I can drop that requirement, to be honest I can't remember why I introduced it in the first place. That said, remember that Swift Package Plugins compatibility is something that is only being run on Xcode 14, which is on beta for now, and that feature is key for this package to accomplish its objective

I dropped the Swift version to 5.5 over at 42c6b6cd0f65269eb3796ff130a968ae21ca01ae. Hope this helps!

daveverwer commented:
Thanks, Jhonatan!

> to be honest I can't remember why I introduced it in the first place.

If you created the package with Xcode 14 beta it will have inserted 5.7 by default!

> I dropped the Swift version to 5.5 over at https://github.com/baguio/SwiftSafeURL/commit/42c6b6cd0f65269eb3796ff130a968ae21ca01ae. Hope this helps!

It does, and the package is now safely in the index. However, 5.6 might be a better choice than 5.5. Swift plugins were introduced with 5.6 so I think that would be a better version to indicate full compatibility. I think the 5.5 indicator is actually what's causing *all* of the compatibility builds to fail, like you can see here:

https://swiftpackageindex.com/baguio/SwiftSafeURL/builds

The decision is entirely yours on which version of the tools to specify, but I think you'll get some better compatibility results using 5.6.

baguio commented:
It's weird that it is not giving me problems with 5.5, but you're correct. I pushed the Swift version back to 5.6, and the build results at SPI are looking just as expected. Thanks for pointing that out
